### PR TITLE
feat: OSCO transformer conform to Results interface class.

### PR DESCRIPTION
**Pull request by degenaro**
Also:
- move timestamping to super class for use by Tanium and OSCO transformers.
- revisions to OSCO transformation itself

## Types of changes

- \[x] Bug fix (non-breaking change which fixes an issue)
- \[x] New feature (non-breaking change which adds functionality)
- \[ \] Breaking change (fix or feature that would cause existing functionality to change)
- \[x My code follows the code style of this project.
- \[ \] My change requires a change to the documentation.
- \[ \] I have updated the documentation accordingly.
- \[ \] I have added tests to cover my changes.
- \[x] All new and existing tests passed.
- \[x] All commits are signed-off.

### CODE REVIEW

**Comment by degenaro** [comment]
Closes #504

File: trestle/transforms/utils/osco_helper.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Facilitate OpenShift Compliance Operator report to NIST OSCAL json transformation."""
+
+import base64
+import bz2
+import logging
+import uuid
+from typing import Any, Dict, List, ValuesView
+from xml.etree import ElementTree
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:22]: See this: https://github.com/IBM/compliance-trestle/issues/535 issue. Not a blocker, however, should be addressed before prod usage.

File: trestle/transforms/utils/osco_helper.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+        prop = ReviewedControls(control_selections=self.control_selections)
+        return prop
+
+    @property
+    def result(self) -> t_result:
+        """OSCAL result."""
+        prop = Result(
+            uuid=str(uuid.uuid4()),
+            title='OpenShift',
+            description='Openshift',
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:304]: Do we want a description that is slightly more descriptive of how this content was produced?

File: trestle/transforms/utils/osco_helper.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+    def _process(self, co_report: ComplianceOperatorReport) -> None:
+        """Process ingested data."""
+        rule_use_generator = co_report.rule_use_generator()
+        try:
+            while True:
+                rule_use = next(rule_use_generator)
+                self._component_extract(rule_use)
+                self._inventory_extract(rule_use)
+                self._observation_extract(rule_use)
+                self._finding_extract(rule_use)
```
Comment (butler54) [inline trestle/transforms/utils/osco_helper.py:418]: As a generator can't you iterate over this using `for rule_use in rule_use_generator` which would avoid the awkward try except block?

**Review by butler54 — APPROVED** [review]
LGTM with the exception of a few 'strange' structures. Have a look  - fix as you feel is appropriate then tag with automerge to pull into the codebase.

File: trestle/transforms/utils/osco_helper.py (hunk 1 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py]: OK, revised to use defusedxml for parse. Continue to use xml.etree.ElementTree.Element for signature typing, which I believe is correct.

File: trestle/transforms/utils/osco_helper.py (hunk 3 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py]: Yep, improved. Thx!

File: trestle/transforms/utils/osco_helper.py (hunk 2 above)
Comment (degenaro) [reply trestle/transforms/utils/osco_helper.py]: Fixed.

**Pull request merged**